### PR TITLE
Making clip above 1 default behavior in preserve_float_range

### LINKED ISSUE #547: Make preserve_float_range have an option to clip values above the upper-bound
Currently, `preserve_float_range` scales `x > 1` to `1` to preserve the data dynamic range. Add an option to allow clipping, for cases where clipping is the preferred behavior.

### PR DESCRIPTION
**Pull request by shanaxel42**
fixes #547

### CODE REVIEW

File: starfish/image/_filter/richardson_lucy_deconvolution.py [hunk 1]
```diff
@@ -130,7 +130,9 @@ def richardson_lucy_deconv(
                 'too many iterations.')
 
         if clip:
-            im_deconv = preserve_float_range(im_deconv)
+            # Changing to cliping values above 1 here changes test results
+            # so keeping this as rescaling for now
+            im_deconv = preserve_float_range(im_deconv, True)
```
Comment (shanaxel42) [inline starfish/image/_filter/richardson_lucy_deconvolution.py]: Ambrose said this was chill

File: starfish/image/_filter/gaussian_high_pass.py [hunk 2]
```diff
@@ -40,9 +40,9 @@ def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
             help="indicates that the image stack should be filtered in 3d")
 
     @staticmethod
-    def high_pass(
-            image: Union[xr.DataArray, np.ndarray], sigma: Union[Number, Tuple[Number]]
-    ) -> Union[xr.DataArray, np.ndarray]:
+    def high_pass(image: Union[xr.DataArray, np.ndarray],
```
Comment (ttung) [inline starfish/image/_filter/gaussian_high_pass.py]: Also, stylistically, I prefer:

```
    def high_pass(
        image: Union[xr.DataArray, np.ndarray],
        sigma: Union[Number, Tuple[Number]],
        rescale: bool = False,
    ) -> Union[xr.DataArray, np.ndarray]:
```

Rationale: lining up the arguments to the opening parenthesis causes a lot of spurious and unnecessary changes when the name of the method changes to something that has a different length.

File: starfish/image/_filter/gaussian_high_pass.py [hunk 3]
```diff
@@ -40,9 +40,9 @@ def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
...
+                  sigma: Union[Number, Tuple[Number]],
+                  rescale: bool = False) -> Union[xr.DataArray, np.ndarray]:
```
Comment (ttung) [inline starfish/image/_filter/gaussian_high_pass.py]: PEP8 says it should be

`rescale: bool=False,`

File: starfish/image/_filter/gaussian_low_pass.py [hunk 4]
```diff
@@ -38,7 +38,8 @@ def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
             help="indicates that the image stack should be filtered in 3d")
 
     @staticmethod
-    def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarray:
+    def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]],
+                 rescale: bool = False) -> np.ndarray:
```
Comment (ttung) [inline starfish/image/_filter/gaussian_low_pass.py]: no spaces around =

File: starfish/image/_filter/richardson_lucy_deconvolution.py (hunk 1 above)
Comment (ttung) [reply starfish/image/_filter/richardson_lucy_deconvolution.py]: LOL

File: starfish/image/_filter/util.py [hunk 5]
```diff
@@ -111,15 +111,20 @@ def validate_and_broadcast_kernel_size(
     return valid_sigma
 
 
-def preserve_float_range(array: Union[xr.DataArray, np.ndarray]) -> Union[xr.DataArray, np.ndarray]:
+def preserve_float_range(
```
Comment (ttung) [inline starfish/image/_filter/util.py]: same thing about lining up to opening parenthesis.

File: starfish/image/_filter/gaussian_low_pass.py [hunk 6]
```diff
@@ -49,11 +53,14 @@ def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarr
...
             Standard deviation of the Gaussian kernel that will be applied. If a float, an
             isotropic kernel will be assumed, otherwise the dimensions of the kernel give (z, y, x)
+         rescale : bool
+            If true scales data by max value, if false clips max values to one
 
         Returns
         -------
         np.ndarray :
             Blurred data in same shape as input image, converted to np.uint16 dtype.
+            :param rescale:
```
Comment (ambrosejcarr) [inline starfish/image/_filter/gaussian_low_pass.py]: Remove this.

File: starfish/image/_filter/gaussian_low_pass.py [hunk 7]
```diff
@@ -49,11 +53,14 @@ def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarr
         sigma : Union[Number, Tuple[Number]]
             Standard deviation of the Gaussian kernel that will be applied. If a float, an
             isotropic kernel will be assumed, otherwise the dimensions of the kernel give (z, y, x)
+         rescale : bool
+            If true scales data by max value, if false clips max values to one
 
         Returns
         -------
         np.ndarray :
             Blurred data in same shape as input image, converted to np.uint16 dtype.
```
Comment (ambrosejcarr) [inline starfish/image/_filter/gaussian_low_pass.py]: No longer uint16. Oops.

File: starfish/image/_filter/gaussian_high_pass.py [hunk 8]
```diff
@@ -52,17 +54,20 @@ def high_pass(
...
         sigma : Union[Number, Tuple[Number]]
             Standard deviation of gaussian kernel
+        rescale : bool
+            If true scales data by max value, if false clips max values to one
 
         Returns
         -------
         np.ndarray :
             filtered image of the same shape as the input image
+            :param rescale:
```
Comment (ambrosejcarr) [inline starfish/image/_filter/gaussian_high_pass.py]: I think your IDE is messing with you.

File: starfish/image/_filter/mean_high_pass.py [hunk 9]
```diff
@@ -68,11 +68,14 @@ def high_pass(image: np.ndarray, size: Number) -> np.ndarray:
...
         size : Number
             width of the kernel
+        rescale : bool
+            If true scales data by max value, if false clips max values to one
 
         Returns
         -------
         np.ndarray :
             Filtered image, same shape as input
+            :param clip:
```
Comment (ambrosejcarr) [inline starfish/image/_filter/mean_high_pass.py]: Here too.

File: starfish/image/_filter/util.py [hunk 10]
```diff
@@ -111,15 +111,20 @@ def validate_and_broadcast_kernel_size(
...
+        array: Union[xr.DataArray, np.ndarray],
+        rescale: bool=False) -> Union[xr.DataArray, np.ndarray]:
     """
-    Clip values below zero and scale values by the max, but only if sub- or super-maximal values are
-    detected. Otherwise, this function will preserve the dynamic range of the input image.
+    Clip values below zero and and above one or scale values by the max,
```
Comment (ambrosejcarr) [inline starfish/image/_filter/util.py]: I think this reads a bit confusingly, since the `or` clause doesn't have a clean antecedent. 

I would rephrase this as 
```
Clips values below zero. If values above one are detected, clips them to 1 unless `rescale` is True, 
in which case the input is scaled by the max value and the dynamic range is preserved. 
```
